### PR TITLE
Feature: Lucky search

### PR DESCRIPTION
**Pull request by maxplt**
# Jira Issue: [IP-290](https://hedvig.atlassian.net/jira/software/projects/IP/boards/19?selectedIssue=IP-290) + [IP-289](https://hedvig.atlassian.net/jira/software/projects/IP/boards/19?selectedIssue=IP-289)

## What?
Add lucky search to new Search and remove some old search components. 

## Why?
For fast navigation 

## Video

https://user-images.githubusercontent.com/89198006/169245025-da792e07-ec4c-4f3c-a661-1a2dee3f9e29.mov

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

### CODE REVIEW

File: src/portals/hope/features/search/members/MemberSearch.tsx [hunk 1]
```diff
@@ -64,6 +64,24 @@ export const MemberSearch: React.FC<{ query: string | null }> = ({
...
+    }
+
+    await search({
+      type: hasSearchCarMembersPermission ? 'CAR_MEMBERS' : 'MEMBERS',
+    })
+
+    if (hits.length) {
+      const luckyHit = hits[0]
+
+      const link = `/members/${luckyHit.hit.memberId}/contracts`
```
Comment (falklinus) [inline src/portals/hope/features/search/members/MemberSearch.tsx:79]:
```suggestion
      const link = `/members/${luckyHit.hit.memberId}`
```

File: src/portals/hope/features/search/members/MemberSearch.tsx [hunk 2]
```diff
@@ -64,6 +64,24 @@ export const MemberSearch: React.FC<{ query: string | null }> = ({
...
+
+      history.push(link)
```
Comment (falklinus) [inline src/portals/hope/features/search/members/MemberSearch.tsx:81]:
```suggestion
      history.push(`/members/${hits[0].hit.memberId}/contracts`)
```
Think this code is clear enough 🧐

Comment (falklinus) [inline src/portals/hope/features/search/members/MemberSearch.tsx:81]:
```suggestion
      history.push(`/members/${hits[0].hit.memberId}/contracts`)
```
Think this code is clear enough 🧐

**Review by vonElfvin — COMMENTED** [review]
Nice! I think the UX can be improved slightly:
1. A less regularly used hotkey for Lucky search
2. Automatically lucky search when there is only one search result

Also I think we're missing keyboard navigation here as it was before. Maybe that can come in a future PR though

**Review by vonElfvin — APPROVED** [review]
🔥

**Pull request merged**